skyportal/tests/fixtures.py

```diff
@@ -530,26 +532,56 @@ def teardown(obj):
         DBSession().delete(obj)
         DBSession().commit()
         for instrument in instruments:
-            InstrumentFactory.teardown(instrument)
+            InstrumentFactory.teardown(instrument.id)
 
 
-class GcnFactory(factory.alchemy.SQLAlchemyModelFactory):
+class GcnNoticeFactory(factory.alchemy.SQLAlchemyModelFactory):
     class Meta(BaseMeta):
-        model = GcnEvent
+        model = GcnNotice
 
     sent_by = factory.SubFactory(UserFactory)
 
     dateobs = datetime.datetime.now()
 
+    ivorn = str(uuid.uuid4())
+
     @staticmethod
-    def teardown(gcn):
-        if is_already_deleted(gcn, GcnEvent):
-            return
+    def teardown(gcnnotice_id):
+        gcnnotice_ = (
+            DBSession()
+            .execute(sa.select(GcnNotice).filter(GcnNotice.id == gcnnotice_id))
+            .scalars()
+            .first()
+        )
+        if gcnnotice_ is not None:
+            # If it is, delete it
+            DBSession().delete(gcnnotice_)
+            DBSession().commit()
```

Comment:
If you are using a new session to tear down, it is even more important not to have this inside another session. 
And I would use a context manager here as well.